libyear-gradle-plugin/src/main/kotlin/com/libyear/traversal/DependencyTraversal.kt

```diff
@@ -27,7 +28,13 @@ class DependencyTraversal private constructor(
         if (maxTransitiveDepth != null && depth > maxTransitiveDepth) {
           continue
         }
+
+        if (excludedPackages.any { selected.moduleVersion.toString().contains(it, ignoreCase = true) }) {
```

Comment:
Probably a drawback of the current "contains string ignore case" logic is that build authors cannot exclude "org.slf4j:slf4j", and include "org.slf4j:slf4j-core", because the former is a substring of the latter (those artefacts do not really exist - I made up an example).

The example might appear contrived, but there are packages out there with confusingly similar names - and eventually build authors might ship extensions to open source packages, just with a suffix appended.

Also to "defend" against future feature requests, I think we have three design alternatives, which might cater better to the expectations of build authors:

### Exact Match

The exclusion list consists of exact matches, and "org.slf4j:slf4j" matches only and exhaustively "org.slf4j:slf4j". Easy win, the list might be a bit harder to maintain, but the end result is very predictable.

### Globbing

We allow to add a wildcard character "\*", to match whole groups of dependencies. For example, ""org.slf4j:slf4j*" would match all dependencies with that prefix. Exclusion of a batches of dependencies is easier, and we can always fall back to exact matches if required.

### Regex

We wield the full power of regex - and then can exclude arbitrary sets of dependencies using a complex pattern. The end result might be less predictable, and writing valid regex is less approachable by less experienced developers.

Can I get your opinion on that? I'd be willing to accept any of above proposals.